app/controllers/service_providers_controller.rb

```diff
@@ -135,6 +134,37 @@ def logo_file_param
     service_provider_params[:logo_file]
   end
 
+  def attach_logo_file
+    return unless logo_file_param
+
+    service_provider.logo_file.attach(logo_file_param)
+    service_provider.logo = service_provider.logo_file.filename.to_s
+    push_logo_content_type
+  end
+
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def push_logo_content_type
```

Comment:
can we add some sort of coverage of this? like a simple unit test that makes sure we call to s3?

Reply:
I'm looking at this and thinking it may be easier to test if we move it into a service

Reply:
Another thing that may be helpful is client stubbing which you get from the AWS SDK: https://aws.amazon.com/blogs/developer/advanced-client-stubbing-in-the-aws-sdk-for-ruby-version-3/

Reply:
Using the client stubbing made the tests much easier :-) I'm going to hold off on refactoring to a service for now.